components/feed/LikeButton.tsx

```diff
@@ -1,60 +1,88 @@
 "use client";
 
-import React, { useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 
-const LikeButton = ({ filled = false }: { filled?: boolean }) => {
-  const [state, setState] = useState(filled);
+import { formatFollowers } from "@/lib/formatters";
+import isItemLiked from "@/lib/likeButton/isItemLiked";
+import likeItem from "@/lib/likeButton/likeItem";
+import unlikeItem from "@/lib/likeButton/unlikeItem";
+
+const LikeButton = ({
+  userId,
+  postId,
+  likes,
+}: {
+  userId: string;
+  postId: string;
+  likes: number;
+}) => {
+  const [liked, setLiked] = useState(false);
```

Comment:
Once you have the likes coming in as a prop, do something like:
```suggestion
  const liked = likedIds.contains(userId);
```

Reply:
that way it's all automatic and doesn't reply on state